services/director/src/simcore_service_director/producer.py

```diff
@@ -534,7 +534,7 @@ async def _get_service_state(
     log.debug("Getting service %s state", service_name)
     tasks = await client.tasks.list(filters={"service": service_name})
 
-    async def _wait_for_tasks(tasks):
+    async def _wait_for_tasks():
```

Comment:
maybe at this point there would be no use for this function

Reply:
cannot use w/o function since it has a return. 
I reverted my change and used the original version with minor changes